pyinteraph/path_analysis.py

```diff
@@ -418,57 +534,67 @@ def main():
         pdb_boolean = True
     
     # Load file, build graphs and get identifiers for graph nodes
-    identifiers, graph = build_graph(fname = args.input_matrix, \
+    identifiers, graph = build_graph(fname = args.input_matrix,
                                      pdb = args.pdb)
 
-    # Convert user input to a list of nodes
-    source_list = convert_input_to_list(user_input = args.source, \
-                                        identifiers = identifiers, \
-                                        pdb = pdb_boolean)
-    target_list = convert_input_to_list(user_input = args.target, \
-                                        identifiers = identifiers, \
-                                        pdb = pdb_boolean)
-    
-    # Choose whether to get shortest paths or all paths
-    if args.do_paths:
-        all_paths = get_all_simple_paths(graph = graph, \
-                              source = source_list, \
-                              target = target_list, \
-                              maxl = args.maxl)
+    # Check if source and target provided
+    if args.source and args.target:
+        # Convert user input to a list of nodes
+        source_list = convert_input_to_list(user_input = args.source,
+                                            identifiers = identifiers,
+                                            pdb = pdb_boolean)
+        target_list = convert_input_to_list(user_input = args.target,
+                                            identifiers = identifiers,
+                                            pdb = pdb_boolean)
+        
+        # Choose whether to get shortest paths or all paths
+        if args.do_paths:
+            all_paths = get_all_simple_paths(graph = graph,
+                                source = source_list,
+                                target = target_list,
+                                maxl = args.maxl)
+            path_type = "all"
+        else:
+            all_paths = get_shortest_paths(graph = graph,
```

Comment:
Check indentation, it should be on the same level as the arguments on the previous lines,